src/validations/validate-parameters.ts

```diff
@@ -0,0 +1,61 @@
+import { Toolkit } from 'actions-toolkit';
+
+const getParametersWithEmptyValues = (tools: Toolkit, schema: Record<any, any>): any => {
+    return Object.entries(schema).reduce<any>((acc, [parameter, parameterConfig]) => {
+        if (!parameterConfig.required) {
+            return acc;
+        }
+
+        if (tools.inputs[parameter]) {
+            return acc;
+        }
+
+        acc.push(parameter);
+        return acc;
+    }, []);
+};
+
+const getParametersWithInvalidValues = (tools: Toolkit, schema: Record<any, any>): any => {
+    return Object.entries(schema).reduce<any>((acc, [parameter, parameterConfig]) => {
+        if (!parameterConfig.availableValues) {
+            return acc;
+        }
+
+        if (!parameterConfig.required && !tools.inputs[parameter]) {
+            return acc;
+        }
+
+        if (!parameterConfig.availableValues.includes(tools.inputs[parameter])) {
+            acc.push(parameter);
+        }
+
+        return acc;
+    }, []);
+};
+
+/**
+ * Validate correct parameters of the action
+ * @param {import('actions-toolkit').Toolkit} tools
+ * @param {object} schema
+ */
+export default async function validateParameters(tools: Toolkit, schema: Record<any, any>): Promise<void> {
```

Comment:
Totally agree in strongly typing the application. I had not done it in this PR as I was planning to do it in successive PRs. But since it is necessary to take advantage of all the potential that Typescript offers us, I have included this typing.